TST: test_time_inputs and test__validate_input_shapes fail in exotic archs

Example log: https://github.com/astropy/astropy/actions/runs/8964757967/job/24617047410

Example test header:

```
  platform linux -- Python 3.12.3, pytest-7.4.4, pluggy-1.4.0
  
  Running tests with Astropy version 7.0.0.dev96+g04fe6537fe.
  Running tests in astropy docs.
  
  Date: 2024-05-06T06:59:10
  
  Platform: Linux-6.5.0-1018-azure-aarch64-with-glibc2.39
  
  Executable: /home/runner/work/astropy/astropy/tests/bin/python3
  
  Full Python Version: 
  3.12.3 (main, Apr 10 2024, 05:33:47) [GCC 13.2.0]
  
  encodings: sys: utf-8, locale: utf-8, filesystem: utf-8
  byteorder: little
  float info: dig: 15, mant_dig: 15
  
  Package versions: 
  Numpy: 1.26.4
  Scipy: not available
  Matplotlib: not available
  h5py: not available
  Pandas: not available
  PyERFA: 2.0.1.4
  Cython: 3.0.10
  Scikit-image: not available
  asdf-astropy: not available
  pyarrow: not available
  
  Using Astropy options: remote_data: none.
  
  CI: true
  ARCH_ON_CI: aarch64
  IS_CRON: true
  
  rootdir: /home/runner/work/astropy/astropy
  configfile: pyproject.toml
  testpaths: astropy, docs
  plugins: xdist-3.6.1, filter-subpackage-0.2.0, arraydiff-0.6.1, doctestplus-1.2.0, mock-3.12.0, cov-4.1.0, remotedata-0.4.1, hypothesis-6.98.15, astropy-0.11.0, astropy-header-0.2.2
```

In `coordinates`:

```
 _______________________________ test_time_inputs _______________________________
  
      def test_time_inputs():
...
          # If the shapes are not broadcastable, then we should raise an exception.
          with pytest.raises(ValueError, match="inconsistent shapes"):
  >           FK4([1, 2, 3] * u.deg, [4, 5, 6] * u.deg, obstime=["J2000", "J2001"])
  
  astropy/coordinates/tests/test_frames.py:732: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  astropy/coordinates/baseframe.py:353: in __init__
      raise exc
  astropy/coordinates/baseframe.py:347: in __init__
      self._shape = np.broadcast_shapes(*shapes)
  numpy/lib/stride_tricks.py:473: in broadcast_shapes
      return _broadcast_shape(*arrays)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  args = (array([(), (), ()], dtype=[]), array([(), ()], dtype=[]))
  
      def _broadcast_shape(*args):
          """Returns the shape of the arrays that would result from broadcasting the
          supplied arrays against each other.
          """
          # use the old-iterator because np.nditer does not handle size 0 arrays
          # consistently
  >       b = np.broadcast(*args[:32])
  E       ValueError: shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (3,) and arg 1 with shape (2,).
  E       non-scalar data and/or attributes with inconsistent shapes: [(3,), (2,)]
  
  numpy/lib/stride_tricks.py:422: ValueError
  
  During handling of the above exception, another exception occurred:
  
      def test_time_inputs():
...    
          # If the shapes are not broadcastable, then we should raise an exception.
  >       with pytest.raises(ValueError, match="inconsistent shapes"):
  E       AssertionError: Regex pattern did not match.
  E        Regex: 'inconsistent shapes'
  E        Input: 'shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (3,) and arg 1 with shape (2,).'
  
  astropy/coordinates/tests/test_frames.py:731: AssertionError
```

In `modeling`:

```
  _________________________ test__validate_input_shapes __________________________
  
      def test__validate_input_shapes():
...
          # Fails because the input shape of the second input has one more axis which
          # for which the first input can be broadcasted to
          inputs = [np.array([[1, 2], [3, 4]]), np.array([[5, 6], [7, 8], [9, 10]])]
          with pytest.raises(ValueError, match=MESSAGE):
  >           model._validate_input_shapes(inputs, model.inputs, 1)
  
  astropy/modeling/tests/test_core.py:998: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  astropy/modeling/core.py:1072: in _validate_input_shapes
      raise exc
  astropy/modeling/core.py:1067: in _validate_input_shapes
      input_shape = np.broadcast_shapes(*all_shapes)
numpy/lib/stride_tricks.py:473: in broadcast_shapes
      return _broadcast_shape(*arrays)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  args = (array([[(), ()],
         [(), ()]], dtype=[]), array([[(), ()],
         [(), ()],
         [(), ()]], dtype=[]))
  
      def _broadcast_shape(*args):
          """Returns the shape of the arrays that would result from broadcasting the
          supplied arrays against each other.
          """
          # use the old-iterator because np.nditer does not handle size 0 arrays
          # consistently
  >       b = np.broadcast(*args[:32])
  E       ValueError: shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (2, 2) and arg 1 with shape (3, 2).
  E       All inputs must have identical shapes or must be scalars.
  
numpy/lib/stride_tricks.py:422: ValueError
  
  During handling of the above exception, another exception occurred:
  
      def test__validate_input_shapes():
...
          # Fails because the input shape of the second input has one more axis which
          # for which the first input can be broadcasted to
          inputs = [np.array([[1, 2], [3, 4]]), np.array([[5, 6], [7, 8], [9, 10]])]
  >       with pytest.raises(ValueError, match=MESSAGE):
  E       AssertionError: Regex pattern did not match.
  E        Regex: 'All inputs must have identical shapes or must be scalars'
  E        Input: 'shape mismatch: objects cannot be broadcast to a single shape.  Mismatch is between arg 0 with shape (2, 2) and arg 1 with shape (3, 2).'
  
  astropy/modeling/tests/test_core.py:997: AssertionError
```

--- a/astropy/coordinates/tests/test_frames.py
+++ b/astropy/coordinates/tests/test_frames.py
@@ -257,21 +257,22 @@ def test_no_data_nonscalar_frames():
     assert a1.temperature.shape == (3, 10)
     assert a1.shape == (3, 10)
 
+    match = r".*inconsistent shapes.*"
     if sys.version_info >= (3, 11) and PYTEST_LT_8_0:
-        # Exception.__notes__ are available (and used here) but ignored in matching,
+        # Exception.__notes__ are available but ignored in matching,
         # so we'll match manually and post-mortem instead
-        match = None
+        direct_match = None
     else:
-        match = r".*inconsistent shapes.*"
+        direct_match = match
 
-    with pytest.raises(ValueError, match=match) as exc:
+    with pytest.raises(ValueError, match=direct_match) as exc:
         AltAz(
             obstime=Time("2012-01-01") + np.arange(10.0) * u.day,
             temperature=np.ones((3,)) * u.deg_C,
         )
 
-    if match is None:
-        assert "inconsistent shapes" in "\n".join(exc.value.__notes__)
+    if direct_match is None:
+        assert re.match(match, "\n".join(exc.value.__notes__))
 
 
 def test_frame_repr():
@@ -728,8 +729,19 @@ def test_time_inputs():
     assert c.shape == (2,)
 
     # If the shapes are not broadcastable, then we should raise an exception.
-    with pytest.raises(ValueError, match="inconsistent shapes"):
+    match = r".*inconsistent shapes.*"
+    if sys.version_info >= (3, 11) and PYTEST_LT_8_0:
+        # Exception.__notes__ are available but ignored in matching,
+        # so we'll match manually and post-mortem instead
+        direct_match = None
+    else:
+        direct_match = match
+
+    with pytest.raises(ValueError, match=direct_match) as exc:
         FK4([1, 2, 3] * u.deg, [4, 5, 6] * u.deg, obstime=["J2000", "J2001"])
+
+    if direct_match is None:
+        assert re.match(match, "\n".join(exc.value.__notes__))
 
 
 def test_is_frame_attr_default():

--- a/astropy/modeling/tests/test_core.py
+++ b/astropy/modeling/tests/test_core.py
@@ -1,6 +1,7 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 # pylint: disable=invalid-name
 import os
+import re
 import subprocess
 import sys
 import unittest.mock as mk
@@ -27,7 +28,7 @@ from astropy.modeling.core import (
 )
 from astropy.modeling.parameters import Parameter
 from astropy.modeling.separable import separability_matrix
-from astropy.tests.helper import assert_quantity_allclose
+from astropy.tests.helper import PYTEST_LT_8_0, assert_quantity_allclose
 from astropy.utils.compat.optional_deps import HAS_SCIPY
 
 
@@ -989,13 +990,24 @@ def test__validate_input_shapes():
     assert (2, 2) == model._validate_input_shapes(inputs, model.inputs, 1)
 
     # Fail check_broadcast
-    MESSAGE = r"All inputs must have identical shapes or must be scalars"
+    match = r".*All inputs must have identical shapes or must be scalars.*"
 
     # Fails because the input shape of the second input has one more axis which
     # for which the first input can be broadcasted to
     inputs = [np.array([[1, 2], [3, 4]]), np.array([[5, 6], [7, 8], [9, 10]])]
-    with pytest.raises(ValueError, match=MESSAGE):
+
+    if sys.version_info >= (3, 11) and PYTEST_LT_8_0:
+        # Exception.__notes__ are available but ignored in matching,
+        # so we'll match manually and post-mortem instead
+        direct_match = None
+    else:
+        direct_match = match
+
+    with pytest.raises(ValueError, match=direct_match) as exc:
         model._validate_input_shapes(inputs, model.inputs, 1)
+
+    if direct_match is None:
+        assert re.match(match, "\n".join(exc.value.__notes__))
 
 
 def test__remove_axes_from_shape():

